src/components/opt-in-out/index.tsx

```diff
@@ -92,13 +98,19 @@ export default function OptInOut({ platform, redirectPath }: OptInOutProps) {
         iconPosition="trailing"
         onClick={openDialog}
       />
-      <Dialog onDismiss={closeDialog} isOpen={showDialog} label="Opt out form">
-        <OptOutForm
-          onSubmit={handleOptOut}
+      {isDialogPreloaded || showDialog ? (
```

Comment:
Open thought: I wonder if a wrapper component would a more ergonomic interface for this?

```
<PreloadNextDynamic show={boolean}>
```

Reply:
What would the implementation / usage look like?

Reply:
Hmm, maybe nevermind! I was thinking we could abstract away the conditionals. But looking at it a bit closer, there's not much net improvement / additional interface is still required...

```diff
- {isDialogPreloaded || showDialog ? (
+ <PreloadNextDynamic component={Dialog}>

- ) : null}
+ </PreloadNextDynamic>
```

Reply:
Ah word gotcha, yeah that could be an option too! I think in conjunction with the hook that would give some nice flexibility.

I think preloading is generally a pretty imperative action, so entrenching it into component/rendering flow isn't always what we want.